### DISCUSSION TITLE
Accessibility of large data sets

### DISCUSSION THREAD

**Discussion by MayteTDGeograma** [gopeg/issues-barriers#31] [General]
The huge size of some original data represents an obstacle to data download and processing. It is highly recommended that apart from a WFS download option, the data providers make their data downloadable in more handy formats such as Geopackages via ATOM feeds. In any case, it is recommended to supply large data divided into multiple files for better manageability.

**Comment by thorsten-reitz** [comment]
A large data set should not be downloaded fully through WFS. Normally, WFS is used to access subsets of large data sets.
For large data sets, there should always be a full dataset download option, such as via atom feed.

**Comment by naomithiru** [comment]
Large datasets have presented a challenge in processing for the Tracing use-case. Full sewer and water datasets with a large number of geometry segments are slow to render on desktop GIS, and slow to load into the preprocessing application. This is true of both shapefiles and gml formats.

**Comment by AndersOstmanNovogit** [comment]
This issue is reported in section 4.2.2 in D4.3

One solution, is to enforce the data provider to publish large data sets using WFS and atom feeds, see comment by Thorsten. The WFS option may then be used to access a smaller portion of the data. Very large atom feeds should also be avoided, since they might create problems at the processing stage. As an example, the INSPIRE validator accepts GML files of maximum 10 Mb and other pieces of software have other limitations. I can from personal experience mention that it is not easy to chunk up a 500 Mb GML file into suitable sizes. Alternative more compact formats like GeoPackage should be considered.
In case the data set can be segmented into smaller portions, a second solution may be to use the smaller segments for exploring the data set and setting up processing paths. Then, the entire data set can be processed as a background process or batch type processing.
In principle, scalable cloud services may also be considered, in case they exist.

Which solutions did you apply?

**Reply by naomithiru** [reply]
In the Tracing use-case, we we worked with a sampled dataset(clipped to a municipal boundary) while developing parts of the preprocessing step, and then eventually with the full datasets. Hale studio is also slow to transform and export transformed data for some large datasets, which could run as a background process after alignment.
(It also helps if the machine on which these datasets are being processed or transformed has a powerful processor.)

However, all the harmonized datasets are in GeoPackage, which has shown much better performance.

**Reply by thorsten-reitz** [reply]
Do you have some specific example about the performance of hale studio? It might get slow if it doesn't have enough memory assigned (by default it uses a maximum of 2GB which can be little if a lot of expensive operations are used, or if a lot of data is loaded into the interactive workbenach instead of using instance sampling)

**Reply by thorsten-reitz** [reply]
> the INSPIRE validator accepts GML files of maximum 10 Mb

Which, from my perspective, is completely impractical. The issue is that many data sets cannot be easily split up in chunks that still validate, such as all networks. IMHO datasets with topology should be evaluated as a complete dataset.

There are several approaches to make (GML) datasets more manageable:
- properly normalise the data, i.e. use by-reference encoding in GML instead of in-place encoding, or by using DatasetDefaults as it was done in the END schema developed with @sMorrone 
- generalise the geometry if it has 10k+ points per object
- reduce unneeded digits from coordinates (no, your data is not precise to one nanometer ;))
- do not create lots of empty, deep, nested elements with minimal information
- provide an additional "core" dataset using a flat structure

**Reply by naomithiru** [reply]
> Do you have some specific example about the performance of hale studio? It might get slow if it doesn't have enough memory assigned (by default it uses a maximum of 2GB which can be little if a lot of expensive operations are used, or if a lot of data is loaded into the interactive workbenach instead of using instance sampling)

The full Flanders sewer dataset was extremely slow on Hale as compared to the other datasets. The sewer edges and sewer nodes have more than 350,000 rows each and a combined file size of around 700mb. (First attempt took more than an hour just to export a GeoPackage). This was observed in both loading the full dataset, in transformation and in exporting transformed data. We used instance sampling for alignment. The solution here was to transform sewer edges separately from the sewer nodes.

**Reply by AndersOstmanNovogit** [reply]
Now in D4.3 section 4.1.1
